File: handlers/wizard_handler.go
package handlers

import (
	"clusterfudge/internal/wizards"
)

// WizardHandler exposes resource creation wizard operations to the frontend.
type WizardHandler struct{}

// NewWizardHandler creates a WizardHandler.
func NewWizardHandler() *WizardHandler {
	return &WizardHandler{}
}

// PreviewDeployment generates a Deployment YAML manifest from the given spec.
func (h *WizardHandler) PreviewDeployment(spec wizards.DeploymentSpec) (string, error) {
	return wizards.DeploymentManifest(spec)
}

// PreviewService generates a Service YAML manifest from the given spec.
func (h *WizardHandler) PreviewService(spec wizards.ServiceSpec) (string, error) {
	return wizards.ServiceManifest(spec)
}

// PreviewConfigMap generates a ConfigMap YAML manifest from the given spec.
func (h *WizardHandler) PreviewConfigMap(spec wizards.ConfigMapSpec) (string, error) {
	return wizards.ConfigMapManifest(spec)
}

// PreviewSecret generates a Secret YAML manifest from the given spec.
func (h *WizardHandler) PreviewSecret(spec wizards.SecretSpec) (string, error) {
	return wizards.SecretManifest(spec)
}

File: ui/src/wailsjs/go/handlers/WizardHandler.ts
import { wailsCall } from '../../call'

const H = 'WizardHandler'

export function PreviewDeployment(spec: Record<string, unknown>): Promise<string> {
  return wailsCall(H, 'PreviewDeployment', spec)
}

export function PreviewService(spec: Record<string, unknown>): Promise<string> {
  return wailsCall(H, 'PreviewService', spec)
}

export function PreviewConfigMap(spec: Record<string, unknown>): Promise<string> {
  return wailsCall(H, 'PreviewConfigMap', spec)
}

export function PreviewSecret(spec: Record<string, unknown>): Promise<string> {
  return wailsCall(H, 'PreviewSecret', spec)
}
